source/lex.tex

```diff
@@ -207,18 +207,17 @@
     \terminal{\textbackslash U} hex-quad hex-quad
 \end{bnf}
 
-The character designated by the \grammarterm{universal-character-name} \tcode{\textbackslash
-U00NNNNNN} is that character
-that has \tcode{U+NNNNNN} as a code point short identifier;
-the character designated by the \grammarterm{universal-character-name}
-\tcode{\textbackslash uNNNN} is that character
-that has \tcode{U+NNNN} as a code point short identifier.
-If a \grammarterm{universal-character-name} does not correspond to
-a code point in ISO/IEC 10646 or
-if a \grammarterm{universal-character-name} corresponds to
-a surrogate code point,
-the program is ill-formed. Additionally, if
-a \grammarterm{universal-character-name} outside
+A \grammarterm{universal-character-name}
+designates the character in ISO/IEC 10646 (if any)
```

Comment:
Can we avoid first talking about "characters" and then retracting our statement later by saying "if any", and then expressly discussing the non-character case?
Maybe saying "... designates a code point given by the hexadecimal number ...", then giving the ill-formed dance, and then continuing with "A code point designates an ISO/IEC ... character, is a reserved code point, or a noncharacter code point. The latter are considered to designate separate characters ..." would help.

Reply:
We use the word "character" in lots of places when talking about sequences of elements from the source character set, which is what a UCN is notionally forming (eg, [lex.name]: "All characters are significant.", [basic]/9: "Two names are the same if they are identifiers composed of the same character sequence", etc.). I don't want to change those all to be talking about sequences of code points, which means we're left with our notion of "character" including things that Unicode doesn't consider to be a character. =(